spec/ddtrace/utils_spec.rb

```diff
@@ -0,0 +1,141 @@
+require 'concurrent-ruby'
+
+RSpec.describe Datadog::Utils do
+  describe '.next_id' do
+    subject(:next_id) { described_class.next_id }
+
+    it 'returns a positive integer smaller than 2**62' do
+      is_expected.to be_a(Integer)
+      is_expected.to be_between(1, 2**62 - 1)
+    end
+
+    it 'fits in a CRuby VALUE slot', if: Datadog::Core::Environment::ObjectSpace.estimate_bytesize_supported? do
+      expect(Datadog::Core::Environment::ObjectSpace.estimate_bytesize(next_id)).to eq(0)
+    end
+
+    it 'returns unique numbers on successive calls' do
+      # This is the same assumption of non-collision we have in our production systems.
+
+      ids = Array.new(100) { described_class.next_id }.uniq
+      expect(ids).to have(100).items
+    end
+
+    it 'returns unique numbers in a threaded environment' do
+      barrier = Concurrent::CyclicBarrier.new(100)
+      threads = Array.new(100) do
+        Thread.new do
+          barrier.wait
+          described_class.next_id
+        end
+      end
+
+      ids = threads.map(&:value).uniq
+      expect(ids).to have(100).items
+    end
+
+    context 'after forking', if: PlatformHelpers.supports_fork? do
+      it 'generates unique ids across forks' do
+        ids = Array.new(3) do
+          _, stdout = expect_in_fork { puts next_id }
+          stdout
+        end.uniq
```

Comment:
Minor: I'm somewhat concerned about possible fragility of this test, e.g. if somehow in a future release of rspec, when we fork and then the process dies, rspec prints something, then this test may become invalid without us realizing it.

My suggestion is adding something here that would break if there was additional output. For instance, by making the expected output json:

```ruby
          _, stdout = expect_in_fork { puts JSON.generate(next_id: next_id) }
          JSON.parse(stdout).fetch("next_id")
```

Reply:
Will do, I'll `to_i` it, it should suffice today.